File: snippets/lesson-2-creation.ts
import { from, range, Observable } from 'rxjs';

const names: string[] = ['Alice', 'Bob', 'Charlie'];
const names$: Observable<string> = from(names);
names$.subscribe((name: string) => console.log(`Name: ${name}`));

const sequence$: Observable<number> = range(1, 5);
sequence$.subscribe((num: number) => console.log(`Number: ${num}`));

File: snippets/lesson-2-creation.cs
using System;
using System.Reactive.Linq;

string[] names = { "Alice", "Bob", "Charlie" };
IObservable<string> namesStream = names.ToObservable();
namesStream.Subscribe((string name) => Console.WriteLine($"Name: {name}"));

IObservable<int> sequenceStream = Observable.Range(1, 5);
sequenceStream.Subscribe((int num) => Console.WriteLine($"Number: {num}"));